Sources/MapboxCoreNavigation/RouteController.swift

```diff
@@ -241,34 +227,46 @@ open class RouteController: NSObject {
     }
     
     /**
-     Asynchronously updates NavNative navigator with the new `RouteProgress`.
-
-     - parameter progress: New route progress to apply to the navigator.
-     - parameter completion: A completion that will be called once the navigator is updated with a boolean indicating
+     Asynchronously updates NavNative navigator with the new `IndexedRouteResponse`.
+     - parameter indexedRouteResponse: New route response to apply to the navigator.
+     - parameter legIndex: A leg index, to start routing from.
+     - parameter completion: A completion that will be called once the navigator is updated indicating
      whether the change was successful.
      */
-    private func updateNavigator(with progress: RouteProgress,
+    private func updateNavigator(with indexedRouteResponse: IndexedRouteResponse,
+                                 fromLegIndex legIndex: Int,
                                  completion: ((Result<RouteInfo, Error>) -> Void)?) {
+        guard case .route(let routeOptions) = indexedRouteResponse.routeResponse.options else {
+            completion?(.failure(RouteControllerError.internalError))
+            return
+        }
         let encoder = JSONEncoder()
-        encoder.userInfo[.options] = progress.routeOptions
-        guard let routeData = try? encoder.encode(progress.route),
+        encoder.userInfo[.options] = routeOptions
+        guard let routeData = try? encoder.encode(indexedRouteResponse.routeResponse),
               let routeJSONString = String(data: routeData, encoding: .utf8) else {
                   completion?(.failure(RouteControllerError.failedToSerializeRoute))
                   return
         }
         
-        let routeRequest = Directions().url(forCalculating: progress.routeOptions).absoluteString
+        let routeRequest = Directions().url(forCalculating: routeOptions).absoluteString
         
         let parsedRoutes = RouteParser.parseDirectionsResponse(forResponse: routeJSONString,
                                                                request: routeRequest, routeOrigin: RouterOrigin.custom)
         if parsedRoutes.isValue(),
-           let route = (parsedRoutes.value as? [RouteInterface])?.first {
-            self.sharedNavigator.setRoutes(route, uuid: sessionUUID, legIndex: UInt32(progress.legIndex)) { result in
+           var routes = parsedRoutes.value as? [RouteInterface],
+           routes.count > indexedRouteResponse.routeIndex {
+            self.sharedNavigator.setMainRoute(routes.remove(at: indexedRouteResponse.routeIndex),
+                                              uuid: sessionUUID,
+                                              legIndex: UInt32(legIndex)) { [weak self] result in
+                self?.sharedNavigator.setAlternativeRoutes(routes,
+                                                           completion: { _ in })
                 completion?(result)
```

Comment:
Is it okay that we don't wait until alternative routes are set and call this callback here?

Reply:
It is. `completion` reports info about the main route, while setting of alternatives is irrelevant to it. It is also does not change the flow if alternatives set failed for some reason.